src/main/java/com/meli/hseixas/listaspartidasfutebol/controller/PartidaController.java

```diff
@@ -12,4 +18,34 @@ public class PartidaController {
     @Autowired
     private PartidaService partidaService;
 
+    @GetMapping
+    public ResponseEntity<List<PartidaDto>> listarTodasAsPartidas(){
+        var todasAsPartidas = partidaService.listarTodasAsPartidas();
+        return ResponseEntity.ok(todasAsPartidas);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<PartidaDto> listarPartidaPorId(@PathVariable Long id){
+        var partidaPorId = partidaService.listarPartidaPorId(id);
+        return ResponseEntity.ok(partidaPorId);
+    }
+
+    @PostMapping
+    public ResponseEntity<PartidaDto> cadastrarPartida(@RequestBody Partida partida, UriComponentsBuilder uriBuilder) {
+        var partidaCadastrada = partidaService.cadastrarPartida(partida);
+        URI uri = uriBuilder.path("/partidas/{id}").buildAndExpand(partidaCadastrada.getId()).toUri();
+        return ResponseEntity.created(uri).body(partidaCadastrada);
```

Comment:
Você pode evitar a necessidade desse URI fazendo assim:

```suggestion
        return new ResponseEntity<PartidaDto>(partidaCadastrada, HttpStatus.CREATED);
```

Reply:
Feito